### ISSUE TITLE
`no-eol-whitespace` cannot fix an inline comment that holds nothing but whitespace under `postcss-scss`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#254]
Found by the third round of review of [#247](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/247), which widened the same rule's reading of where a line ends. Not that reading's doing: the shape behaves this way for every spelling of the break, and has since the rule learned to fix a comment at all.

### What minimal example or steps are needed to reproduce the bug?

An inline comment holding nothing but whitespace, in a Sass file:

```scss
//  
b {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/no-eol-whitespace": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

`Unexpected whitespace at end of line` at `1:4`, and the two spaces trimmed, so that the file comes back as `//\nb {}` and a second run has nothing to say. That is what the same file gets under `postcss-less`, and what a comment holding text gets under either syntax: `// c  \n` is reported at `1:6` and comes back `// c\n`.

### What actually happened?

The warning is reported and the file comes back byte-identical, so `--fix` ends on a clean pass over a stylesheet it did not touch and the warning goes with it. Every run reports the same thing and no run changes anything. All four spellings of the break behave alike — `//  \n`, `//  \r`, `//  \r\n` and `//  \f` — and so does a comment standing anywhere in the file rather than at its end.

### What is the cause?

`postcss-scss` files a whitespace-only inline comment as `text: ""` with every character of the whitespace in `raws.left`, and leaves the break outside the node. The fix of [`no-eol-whitespace`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/no-eol-whitespace/index.js) walks the lines of `raws.left`, which trims the whitespace standing in front of a break **inside** that raw — and there is none — while the trim of a raw's very end is applied to `raws.right`, which is empty for such a comment. So nothing reaches the two spaces.

`postcss-less` is affected too, and not where this reads first. It closes an inline comment on a line feed and on nothing else, so a bare carriage return or a form feed leaves the rest of the file as the comment's own text — `//  \rb {}` comes back as `text: "b {}"`, `left: "  \r"` — while at `\n` and `\r\n` the whitespace goes to the `raws.before` of the node behind it, or to the root's own `raws.after` where none stands there. The walk over the lines of those raws reaches all of that.

What it does not reach is the same shape Sass has: a comment holding no text at all, which under Less is what a file ending inside one spells. `//  `, `//\r  ` and `//  \f  ` are all reported and all come back byte-identical, however many breaks the comment holds. So the repair below answers both syntaxes rather than Sass alone.

Repairing it means trimming `raws.left` wherever an inline comment holds no text at all — a block comment holding nothing but whitespace wears the very same shape, `/*  */` giving `text: ""` and `left: "  "` too, and its whitespace stands in front of the closing delimiter rather than at the end of a line. That changes what `--fix` writes for an ordinary line-feed file, which is why [#247](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/247) named the shape and left it standing rather than taking it on.
